code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm

```diff
@@ -6,113 +6,163 @@
+	var/mob/living/carbon/xenomorph/X = owner
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!isxeno(target) || target.hive.hivenumber != X.hive.hivenumber)
+		return FALSE
+	if(!X.Adjacent(target))
+		X.balloon_alert(X, "Not adjacent")
+		return FALSE
+	if(!existing_link && HAS_TRAIT(target, TRAIT_ESSENCE_LINKED))
+		target.balloon_alert(X, "She is already linked")
+		return FALSE
+	if(existing_link && target != linked_target)
+		target.balloon_alert(X, "Not our linked sister")
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/essence_link/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/X = owner
+	if(!existing_link)
+		target.balloon_alert(X, "Linking...")
+		if(!do_after(X, DRONE_ESSENCE_LINK_DELAY, TRUE, target, BUSY_ICON_FRIENDLY, BUSY_ICON_FRIENDLY))
+			X.balloon_alert(X, "Link cancelled")
+			return
+		X.apply_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK, 1, target)
+		existing_link = X.has_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+		linked_target = target
+		target.balloon_alert(target, "Essence Link established")
+	succeed_activate()
+
+/datum/action/xeno_action/activable/essence_link/alternate_action_activate()
+	var/mob/living/carbon/xenomorph/X = owner
+	existing_link = X.has_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+	if(!existing_link)
+		X.balloon_alert(X, "No link to cancel")
+		return
+	end_ability()
+	return COMSIG_KB_ACTIVATED
+
+/// Ends the ability, removing signals and buffs.
+/datum/action/xeno_action/activable/essence_link/proc/end_ability()
+	var/mob/living/carbon/xenomorph/X = owner
+	var/datum/action/xeno_action/enhancement/enhancement_action = X.actions_by_path[/datum/action/xeno_action/enhancement]
+	enhancement_action.end_ability()
+	X.remove_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+	existing_link = null
+	linked_target = null
+	add_cooldown()
+
+/datum/action/xeno_action/activable/essence_link/update_button_icon()
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/actions.dmi', button, "essence_link_[existing_link ? (existing_link.stacks) : (0)]")
+	return ..()
+
+// ***************************************
+// *********** Acidic Salve
 // ***************************************
 /datum/action/xeno_action/activable/psychic_cure/acidic_salve
 	name = "Acidic Salve"
 	action_icon_state = "heal_xeno"
-	mechanics_text = "Slowly heal an ally with goop. Apply repeatedly for best results."
+	mechanics_text = "Apply a minor heal to the target. If applied to a linked sister, it will also apply a regenerative buff. Additionally, if that linked sister is near death, the heal's potency is tripled"
 	cooldown_timer = 5 SECONDS
 	plasma_cost = 150
 	keybind_signal = COMSIG_XENOABILITY_PSYCHIC_CURE
 	heal_range = DRONE_HEAL_RANGE
 	target_flags = XABB_MOB_TARGET
+	/// Amount of health restored by this ability.
+	var/heal_amount = 50
+	/// Multiplier applied to this heal when below a certain threshold.
+	var/heal_multiplier = 1
```

Comment:
Does this need to be declared in the action def instead of in the salve_healing proc? Also, as is you'll have a 3x multi forever after you crit heal someone once.

Reply:
Not really, moved it and fixed the issue.